mechanics/tools/gdocs-migrate/main.go

```diff
@@ -26,28 +28,49 @@ func main() {
 	}
 
 	drives := map[string]string{
-		"Knative Community [old]": KnativeCommunityDriveId,
-		"Knative Team":            KnativeTeamDriveId,
+		"Knative Community Drive": KnativeCommunityDriveID,
+		"Knative Team Drive":      KnativeTeamDriveID,
 	}
 
+	forest := NewForest()
 	for name, id := range drives {
-		// List files from Knative Community Drive.
-		r, err := srv.Files.List().PageSize(10).
-			Corpora("drive").
-			IncludeItemsFromAllDrives(true).
-			SupportsAllDrives(true).
-			DriveId(id).
-			Fields("nextPageToken, files(id, name)").Do()
-		if err != nil {
-			log.Fatal("Unable to retrieve files:", err)
-		}
-		fmt.Printf("---------------------------\n%s files:\n---------------------------\n", name)
-		if len(r.Files) == 0 {
-			fmt.Println("No files found.")
-		} else {
+		fmt.Println("Loading directory from ... ", name)
+		forest.Add(File{
+			id:       id,
+			name:     name,
+			isFolder: true,
+		})
+		var pageToken *string
+		for pageToken == nil || *pageToken != "" {
+			// List files from Knative Community Drive.
+			request := srv.Files.List().PageSize(1000).
+				Corpora("drive").
+				IncludeItemsFromAllDrives(true).
+				SupportsAllDrives(true).
+				DriveId(id).
+				Fields("nextPageToken, files(id, name, parents, mimeType)").
+				OrderBy("folder,modifiedTime desc,name")
+			if pageToken != nil {
+				request.PageToken(*pageToken)
+			}
+			r, err := request.Do()
+			if err != nil {
+				log.Fatal("Unable to retrieve files:", err)
+			}
+			pageToken = &r.NextPageToken
 			for _, i := range r.Files {
-				fmt.Printf("%s (%s)\n", i.Name, i.Id)
+				forest.Add(File{
+					id:       i.Id,
+					name:     i.Name,
+					parentID: i.Parents[0],
+					isFolder: i.MimeType == FolderMimeType,
+				})
 			}
+			fmt.Printf("Loaded %d files, %d folders\n", len(forest.GetFiles()), forest.FolderCount())
```

Comment:
I'm assuming this works as a sort of progress bar when inside this loop?

Reply:
Yes, exactly..